src/main/java/tp/cap5buddy/contacts/ContactList.java

```diff
@@ -0,0 +1,91 @@
+package tp.cap5buddy.contacts;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import tp.cap5buddy.contacts.exceptions.ContactNotFoundException;
+import tp.cap5buddy.contacts.exceptions.DuplicateContactException;
+
+public class ContactList {
+
+    public final List<Contact> contacts;
+
+    public ContactList(List<Contact> contacts) {
+        this.contacts = contacts;
+    }
+
+    /**
+     * Returns true if the list contains an equivalent person as the given argument.
+     */
+    public boolean hasContact(Contact toCheck) {
+        requireNonNull(toCheck);
+        return contacts.stream().anyMatch(toCheck::isSameContact);
+    }
+
+    /**
+     * Adds a person to the list.
+     * The person must not already exist in the list.
+     */
+    public void addContact(Contact toAdd) {
+        requireNonNull(toAdd);
+        if (hasContact(toAdd)) {
+            // throw new DuplicatePersonException();
+        }
+        contacts.add(toAdd);
+    }
```

Comment:
Will it work? Since this.contacts is final.

Reply:
Same issue as above.